geniza/corpus/tests/test_corpus_models.py

```diff
@@ -139,7 +139,10 @@ def test_str(self):
         frag = Fragment.objects.create(shelfmark='Or.1081 2.25')
         doc = Document.objects.create()
         doc.fragments.add(frag)
-        assert str(doc) == doc.shelfmark
+        assert doc.shelfmark in str(doc) and str(doc.id) in str(doc)
+
+        unsaved_doc = Document()
+        assert str(unsaved_doc) == ' (??)'
```

Comment:
Is this extra space okay? Or should I add the extra logic to work around this?

Reply:
Right, no shelfmark either when unsaved... what about `?? (PGPID ??)` for unsaved object?